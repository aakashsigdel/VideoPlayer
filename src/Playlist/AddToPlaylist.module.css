.dialog {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 400px;
  height: 380px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2), 0px 24px 38px 3px rgba(0,0,0,0.14), 0px 9px 46px 8px rgba(0,0,0,0.12);
  background-color: #fff;
  border-radius: 4px;
}

.dialog:focus {
  outline: none;
}

.overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
}

.root {
  display: flex;
  flex-direction: column;
  height: inherit;
  padding: 24px 24px 8px;
  box-sizing: border-box;
}

.form {
  flex: 1;
  display: flex;
  flex-direction: column;
}

.formContent {
  flex: 1
}

.footer {
  display: flex;
  justify-content: flex-end;
}
