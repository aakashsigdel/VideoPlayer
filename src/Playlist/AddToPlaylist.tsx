import * as React from 'react'
import Dialog from 'react-modal'

import { IVideo } from '../VideoPlayer'
import { Button } from '../commons/Button'
import { TextField } from '../commons/TextField'

import styles from './AddToPlaylist.module.css'

export interface IAddToPlaylistState extends IVideo {}

export interface IAddToPlaylistProps {
  onClickAdd: (video: IVideo) => void
  onClickCancel: () => void
}

export class AddToPlaylist extends React.Component<
  IAddToPlaylistProps,
  IAddToPlaylistState
> {
  state = {
    artist: '',
    title: '',
    videoUrl: '',
  }

  handleOnSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
    const { onClickAdd } = this.props
    e.preventDefault()
    onClickAdd(this.state)
  }

  handleOnChage = (type: keyof IAddToPlaylistState) => (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    this.setState({ [type]: e.target.value } as Pick<
      IAddToPlaylistState,
      keyof IAddToPlaylistState
    >) // this is a typescript limitation https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635
  }

  render() {
    const { onClickAdd, onClickCancel } = this.props
    return (
      <Dialog
        isOpen={true}
        onRequestClose={onClickCancel}
        contentLabel="Example Modal"
        className={styles.dialog}
        overlayClassName={styles.overlay}
        shouldCloseOnOverlayClick={true}
        ariaHideApp={false}
      >
        <section className={styles.root}>
          <header>
            <h2>Add to Playlist</h2>
          </header>
          <form onSubmit={this.handleOnSubmit} className={styles.form}>
            <div className={styles.formContent}>
              <TextField
                label="Artist"
                type="text"
                id="artist"
                onChange={this.handleOnChage('artist')}
              />
              <TextField
                label="Title"
                type="text"
                id="title"
                onChange={this.handleOnChage('title')}
              />
              <TextField
                label="Video URL"
                type="url"
                id="videoUrl"
                onChange={this.handleOnChage('videoUrl')}
              />
            </div>
            <footer className={styles.footer}>
              <Button
                type="button"
                onClick={onClickCancel}
                label="Cancel"
                variant="raised"
              />
              <Button type="submit" label="Add" variant="raised" />
            </footer>
          </form>
        </section>
      </Dialog>
    )
  }
}
