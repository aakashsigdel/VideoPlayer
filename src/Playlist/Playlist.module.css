.root {
  width: 350px;
  height: 348px;
  margin: 0px 16px;
  font-size: 12px;
  font-weight: bold;
  color: rgba(0, 0, 0, 0.75);
}

.listHead {
  display: flex;
  padding: 0px 4px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.35);
  color: rgba(0, 0, 0, 0.35);
}

.list {
  padding: 0px;
  margin: 0px;
  height: 315px;
  overflow-y: scroll;
}

.listItem {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  text-decoration: none;
  width: 100%;
  text-align: left;
  padding: 10px 4px;
  cursor: pointer;
}

.listItem:hover {
  background-color: rgba(0, 0, 0, 0.08);
}

.number {
  flex: 1;
  color: lightgray;
}

.title {
  flex: 5;
  padding-right: 2px;
}

.artist {
  flex: 5;
  padding-right: 2px;
}

.ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
