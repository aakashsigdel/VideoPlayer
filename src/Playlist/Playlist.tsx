import * as React from 'react'

import { IVideo } from '../VideoPlayer'
import { Button } from '../commons/Button'

import styles from './Playlist.module.css'

export interface IPlaylistProps {
  videos: IVideo[]
  children: (video: IVideo, videoNumber: number) => JSX.Element
  onClickAdd: () => void
}

export const Playlist: React.FunctionComponent<IPlaylistProps> = ({
  videos,
  children,
  onClickAdd,
}) => (
  <section className={styles.root}>
    <div className={styles.listHead}>
      <span className={styles.number}>#</span>
      <span className={styles.title}>Title</span>
      <span className={styles.artist}>Artist</span>
    </div>
    <ul className={styles.list}>
      {videos.map(children)}
      {videos.map(children)}
      {videos.map(children)}
    </ul>
    <Button
      onClick={onClickAdd}
      label="Add to playlist"
      variant="raised"
      type="button"
      className={styles.button}
    />
  </section>
)
