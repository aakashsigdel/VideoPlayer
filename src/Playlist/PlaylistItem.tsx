import * as React from 'react'
import cn from 'classnames'

import { Play, Pause } from '../icons'
import { IVideo } from '../VideoPlayer'
import styles from './Playlist.module.css'

export interface IPlaylistItemProps {
  video: IVideo
  videoNumber: number
  onSelectVideo: (videoNumber: number) => void
  nowPlaying: boolean
}

export const PlaylistItem: React.FunctionComponent<IPlaylistItemProps> = ({
  video,
  videoNumber,
  onSelectVideo,
  nowPlaying,
}) => (
  <li
    onClick={() => onSelectVideo(videoNumber)}
    className={cn(styles.listItem, { [styles.nowPlaying]: nowPlaying })}
  >
    <span className={styles.number}>{videoNumber + 1}</span>
    <span className={cn(styles.title, styles.ellipsis)}>{video.title}</span>
    <span className={cn(styles.artist, styles.ellipsis)}>{video.artist}</span>
  </li>
)
