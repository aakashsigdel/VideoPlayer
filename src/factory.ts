export const getSong = (suffix: string = '1') => ({
  title: `testSong${suffix}`,
  artist: `testArtist${suffix}`,
  videoUrl: `testUrl${suffix}`,
})

export const getSongs = () => {
  return ['1', '2', '3'].map(getSong)
}
