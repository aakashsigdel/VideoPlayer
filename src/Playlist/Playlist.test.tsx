import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow, ShallowWrapper } from 'enzyme'

import { Playlist, IPlaylistProps } from './Playlist'
import { getVideos } from '../factory'

describe('<Playlist />', () => {
  let wrapper: ShallowWrapper<IPlaylistProps>
  const defaultProps: IPlaylistProps = {
    videos: getVideos(),
    onClickAdd: jest.fn(),
    children: (s, k) => <div key={k} />,
  }

  beforeEach(() => {
    wrapper = shallow(<Playlist {...defaultProps} />)
  })

  it('should render correctly', () => {
    const tree = create(<Playlist {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call onClickAdd when clickin on add to playlist button', () => {
    wrapper.find('button').simulate('click')
    expect(defaultProps.onClickAdd).toHaveBeenCalledTimes(1)
  })
})
