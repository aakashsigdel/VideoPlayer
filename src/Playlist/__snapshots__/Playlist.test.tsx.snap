// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playlist /> should render correctly 1`] = `
<section>
  <ul>
    <div />
    <div />
    <div />
  </ul>
  <button
    onClick={[MockFunction]}
  >
    Add to playlist
  </button>
</section>
`;
